drivers/sensor/lsm6dsl/lsm6dsl_shub.c

```diff
@@ -53,23 +53,23 @@ static int lsm6dsl_shub_write_slave_reg(struct lsm6dsl_data *data,
 #define LIS2MDL_OFF_CANC          0x02
 #define LIS2MDL_SENSITIVITY       1500
 
-static int lsm6dsl_lis2mdl_init(struct lsm6dsl_data *data, uint8_t i2c_addr)
+static int lsm6dsl_lis2mdl_init(const struct device *dev, uint8_t i2c_addr)
 {
 	uint8_t mag_cfg[2];
 
 	data->magn_sensitivity = LIS2MDL_SENSITIVITY;
```

Comment:
Leftover ?

Reply:
@erwango 
I just wanted to minimize the changes. But your suggestion looks reasonable.
I'll do the change today